workspaces/homepage/plugins/dynamic-home-page/src/hooks/useTemplates.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+
+import { useApi } from '@backstage/core-plugin-api';
+import useAsyncRetry from 'react-use/lib/useAsyncRetry';
+
+import { modelCatalogApiRef } from '../api';
+
+export const useTemplates = () => {
+  const [isLoading, setIsLoading] = React.useState<boolean>(true);
+  const [data, setData] = React.useState<any>();
```

Comment:
please add the newly added template type here as well.
```suggestion
  const [data, setData] = React.useState<QueryEntitiesResponse<Template>>();
```